File: app/src/main/java/com/byd/dashcast/proxy/ShellGatewayRoutingPolicy.java
package com.byd.dashcast.proxy;

/** Pure routing decision for the proxy/Binder and local-ADB shell paths. */
final class ShellGatewayRoutingPolicy {
    enum Route { PROXY, LEGACY, FAIL_FAST }

    private ShellGatewayRoutingPolicy() {}

    static Route select(boolean legacyPath, boolean proxyConnected,
                        boolean transportUnreachable) {
        if (!legacyPath && proxyConnected) return Route.PROXY;
        return transportUnreachable ? Route.FAIL_FAST : Route.LEGACY;
    }
}

File: app/src/main/java/com/byd/dashcast/proxy/ShellGatewayRoutingPolicy.kt
package com.byd.dashcast.proxy

/** Pure routing decision for the proxy/Binder and local-ADB shell paths. */
object ShellGatewayRoutingPolicy {
    enum class Route { PROXY, LEGACY, FAIL_FAST }

    @JvmStatic
    fun select(legacyPath: Boolean, proxyConnected: Boolean, transportUnreachable: Boolean): Route {
        if (!legacyPath && proxyConnected) return Route.PROXY
        return if (transportUnreachable) Route.FAIL_FAST else Route.LEGACY
    }
}
